src/targets/__tests__/maven.test.ts

```diff
@@ -0,0 +1,174 @@
+    mavenSettingsPath: DEFAULT_OPTION_VALUE,
+    mavenRepoId: DEFAULT_OPTION_VALUE,
+    mavenRepoUrl: DEFAULT_OPTION_VALUE,
+    android: {
+      distDirRegex: '/distDir/',
+      fileReplaceeRegex: '/replacee/',
+      fileReplacerStr: 'replacer',
+    },
+  };
+}
+
+function createMavenTarget(targetOptions?: MavenTargetConfig): MavenTarget {
+  const finalOptions = targetOptions ? targetOptions : getTargetOptions();
+  const options = {
+    name: 'maven',
+    ...finalOptions,
+  };
+  return new MavenTarget(options, new NoneArtifactProvider());
+}
+
+describe('Maven target configuration', () => {
+  beforeEach(() => removeTargetSecretsFromEnv());
+
+  function getExpectedValueOfConfigKey(configKey: string): string | RegExp {
+    if (
+      configKey.includes('androidDistDirPattern') ||
+      configKey.includes('androidFileSearchPattern')
+    ) {
+      return new RegExp(DEFAULT_OPTION_VALUE);
+    }
+    return DEFAULT_OPTION_VALUE;
+  }
+
+  test('with options', () => {
+    setTargetSecretsInEnv();
+    const mvnTarget = createMavenTarget(getTargetOptions());
+    targetOptions.map(secret =>
+      expect(mvnTarget.config).toEqual(
+        expect.objectContaining({
+          [secret]: getExpectedValueOfConfigKey(secret),
+        })
+      )
+    );
+  });
+
+  test('without options', () => {
+    expect(createMavenTarget).toThrowError(ConfigurationError);
+  });
+});
+
+describe('publish to Maven', () => {
+  beforeAll(() => setTargetSecretsInEnv());
+
+  afterAll(() => removeTargetSecretsFromEnv());
+
+  beforeEach(() => jest.resetAllMocks());
+
+  test('main flow', async () => {
+    const mvnTarget = createMavenTarget(getTargetOptions());
+    const gradlePropsMock = jest.fn();
+    mvnTarget.createUserGradlePropsFile = gradlePropsMock;
+    const uploadMock = jest.fn();
+    mvnTarget.upload = uploadMock;
+    const closeAndReleaseMock = jest.fn();
+    mvnTarget.closeAndRelease = closeAndReleaseMock;
+
+    const version = '1.0.0';
+    const revision = 'r3v1s10n';
+
+    await mvnTarget.publish(version, revision);
+    expect(gradlePropsMock).toHaveBeenCalledTimes(1);
+    expect(uploadMock).toHaveBeenCalledTimes(1);
+    expect(uploadMock).toHaveBeenLastCalledWith(revision);
+    expect(closeAndReleaseMock).toHaveBeenCalledTimes(1);
+  });
+
+  test('upload', async () => {
+    const mvnTarget = createMavenTarget(getTargetOptions());
+    mvnTarget.getArtifactsForRevision = jest
+      .fn()
+      .mockResolvedValueOnce([{ filename: 'mockArtifact.zip' }]);
+    mvnTarget.artifactProvider.downloadArtifact = jest
+      .fn()
+      .mockResolvedValueOnce('artifact/download/path');
+
+    await mvnTarget.upload('r3v1s10n');
+    expect(retrySpawnProcess).toBeCalledTimes(1);
+  });
+
+  test('close and release', async () => {
+    const mvnTarget = createMavenTarget(getTargetOptions());
+    await mvnTarget.closeAndRelease();
+    expect(retrySpawnProcess).toHaveBeenCalledTimes(1);
+  });
```

Comment:
This is just testing the implementation, not a very useful test. I'd rather remove this case.